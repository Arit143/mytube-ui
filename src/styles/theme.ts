import { createMuiTheme } from "@material-ui/core/styles";
import blue from "@material-ui/core/colors/blue";

/**
 * Material UI theme creator
 */
export const theme = createMuiTheme({
  palette: {
    primary: blue
  }
});
