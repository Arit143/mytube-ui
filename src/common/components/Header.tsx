import React from "react";
import { Link } from "react-router-dom";

import BottomNavigation from "@material-ui/core/BottomNavigation";
import BottomNavigationAction from "@material-ui/core/BottomNavigationAction";
import { makeStyles } from "@material-ui/styles";
import { Theme } from "@material-ui/core/styles/createMuiTheme";

import List from "@material-ui/icons/List";
import Publish from "@material-ui/icons/Publish";

import { routes } from "./../../routes";

const useStyles = makeStyles((theme: Theme) => {
  return {
    root: {
      color: theme.palette.primary.dark
    },
    nav: {
      background: theme.palette.primary.contrastText
    }
  };
});

/**
 * The header bar for showing the list of videos and upload the videos
 */
const Header: React.FC = () => {
  const classes = useStyles();
  return (
    <BottomNavigation showLabels className={classes.nav}>
      <BottomNavigationAction
        component={Link}
        to={routes[0].path}
        label="List of Videos"
        value="list"
        icon={<List />}
        className={classes.root}
      />
      <BottomNavigationAction
        component={Link}
        to={routes[1].path}
        label="Upload Video"
        value="upload"
        icon={<Publish />}
        className={classes.root}
      />
    </BottomNavigation>
  );
};

export default Header;
