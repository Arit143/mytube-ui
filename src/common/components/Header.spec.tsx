import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "@material-ui/styles";
import { HashRouter as Router, Route } from "react-router-dom";

import Header from "./Header";
import { theme } from "../../styles/theme";

it("Header should match snapshot", () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>
          <Header />
        </Router>
      </ThemeProvider>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
