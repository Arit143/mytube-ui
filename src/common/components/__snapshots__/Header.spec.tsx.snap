// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div
  className="MuiBottomNavigation-root makeStyles-nav-2"
>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root makeStyles-root-1"
    href="#/"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label"
      >
        List of Videos
      </span>
    </span>
  </a>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiBottomNavigationAction-root makeStyles-root-1"
    href="#/upload"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label"
      >
        Upload Video
      </span>
    </span>
  </a>
</div>
`;
